SCSH_3C_iOS/SCSH_3C_iOS/Modules/Explore/View/BoardSectionCollectionViewCell.swift

```diff
@@ -0,0 +1,202 @@
+        }
+        
+        collectionView.setContentOffset(.init(x: contentOffset, y: collectionView.contentOffset.y) , animated: true)
+    }
+    
+    private func setupRecognizer() {
+        collectionView.addGestureRecognizer(hoverRecognizer)
+        hoverRecognizer.addTarget(self, action: #selector(hoverRecognizerDidTrigger(_:)))
+    }
+    
+    @objc
+    private func hoverRecognizerDidTrigger(_ recognizer: UIHoverGestureRecognizer) {
+        switch recognizer.state {
+        case .began, .changed:
+            previousButton.alpha = 0.3
+            nextButton.alpha = 0.3
+            break
+        case .ended, .cancelled:
+            previousButton.alpha = 0
+            nextButton.alpha = 0
+            break
+        default:
+            break
+        }
+    }
+    
+    func setupInfiniteScroll() {
+        // get the first item from `items` where close to the center in the infinite scroll UICollectionView.
+        let firstItem = (infiniteScrollItems / 2) - ((infiniteScrollItems / 2) % items.count)
+        collectionView.scrollToItem(at: IndexPath(item: firstItem, section: 0), at: .centeredHorizontally, animated: false)
+    }
+}
+
+extension BoardSectionCollectionViewCell: UICollectionViewDelegateFlowLayout {
+    func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        let itemWidth = collectionView.frame.width
+        let currentItem = round(collectionView.contentOffset.x / itemWidth)
+        
+        // update page
+        pageControl.currentPage = Int(currentItem) % items.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        infiniteScrollItems
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        CGSize(width: collectionView.bounds.width, height: collectionView.bounds.height)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        0
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        // TODO: implement it.
+        print("BoardSectionCollectionViewCell didSelectItemAt \(indexPath)")
+    }
+}
+
+extension BoardSectionCollectionViewCell: UICollectionViewDataSource {
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "BoardCollectionViewCell", for: indexPath) as! BoardCollectionViewCell
+
+        // TODO: implement it.
+        if indexPath.item % items.count == 0 {
+            cell.contentView.backgroundColor = .red
+        } else {
+            cell.contentView.backgroundColor = .black
+        }
```

Comment:
I'm not entirely certain if this is a widely used convention in Swift, but in my view

```swift
cell.contentView.backgroundColor = .black
if indexPath.item % items.count == 0 {
    cell.contentView.backgroundColor = .red
}
```
appears more favorable

Reply:
Your version is more concise, I'll update it to yours.

Reply:
[done](https://github.com/No-57/SCSH_3C_iOS/pull/27/commits/4b084cdbf5568d1ef940c185f9d2dfdbebccee62)